approov-service/src/main/java/io/approov/service/okhttp/ApproovService.java

```diff
@@ -862,6 +903,69 @@ else if (approovResults.getStatus() != Approov.TokenFetchStatus.UNKNOWN_KEY)
             }
         }
 
+        // if message signing is enabled, add the signature header to the request
+        if (messageSigningConfigs != null) {
+            // Match the domain of the request URL against the configured message signing domains
+            String domain = request.url().host();
+            ApproovMessageSigningConfig messageSigningConfig = messageSigningConfigs.get(domain);
+            if (messageSigningConfig == null) {
+                // If no specific configuration is found for the domain, use the default configuration
+                domain = "*";
+                messageSigningConfig = messageSigningConfigs.get(domain);
+            }
+            if (messageSigningConfig == null) {
+                // If no default configuration is found, do not sign the message
+                return chain.proceed(request);
+            }
+            Log.d(TAG, "Signing message with configuration for domain: "  + domain + ":  " + (messageSigningConfig.getSignApproovToken() ? ", " + approovTokenHeader +  " **** " : "") + " headers " + messageSigningConfig.getSignedHeaders() +
```

Comment:
This code should all move into the "public String generateMessage(request Request, String approovTokenHeader);" in the DefaultMessageSigningConfig implementation.